wp-content/plugins/wporg-learn/inc/blocks.php

```diff
@@ -142,6 +145,115 @@ function lesson_plan_details_render_callback( $attributes, $content ) {
 	return ob_get_clean();
 }
 
+/**
+ * Register Course Data block type and related assets.
+ *
+ * @throws Error If the build files are not found.
+ */
+function register_course_data() {
+	$script_asset_path = get_build_path() . 'course-data.asset.php';
+	if ( ! is_readable( $script_asset_path ) ) {
+		throw new Error(
+			'You need to run `npm start` or `npm run build` for the "wporg-learn/course-data" block first.'
+		);
+	}
+
+	$script_asset = require $script_asset_path;
+	wp_register_script(
+		'course-data-editor-script',
+		get_build_url() . 'course-data.js',
+		$script_asset['dependencies'],
+		$script_asset['version'],
+		true,
+	);
+
+	wp_register_style(
+		'course-data-style',
+		get_build_url() . 'style-course-data.css',
+		array(),
+		filemtime( get_build_path() . 'style-course-data.css' )
+	);
+
+	register_block_type( 'wporg-learn/course-data', array(
+		'editor_script'   => 'course-data-editor-script',
+		'style'           => 'course-data-style',
+		'render_callback' => __NAMESPACE__ . '\course_data_render_callback',
+	) );
+}
+
+/**
+ * Render the block content (html) on the frontend of the site.
+ *
+ * @param array  $attributes
+ * @param string $content
+ * @return string HTML output used by the block
+ */
+function course_data_render_callback( $attributes, $content ) {
+	if ( get_post_type() !== 'course' ) {
+		return;
+	}
+
+	$course_service = new Sensei_Reports_Overview_Service_Courses();
+	$post           = get_post();
+	$course_id      = $post->ID;
+
+	// Get the total number of learners enrolled in the course
+	$learners = Sensei_Utils::sensei_check_for_activity(
+		array(
+			'type'     => 'sensei_course_status',
+			'status'   => 'in-progress',
+			'post__in' => $course_id,
+		)
+	);
+
+	/**
+	// Get the total number of learners who have completed the course
+	$completions = Sensei_Utils::sensei_check_for_activity(
+		array(
+			'type'     => 'sensei_course_status',
+			'status'   => 'complete',
+			'post__in' => $course_id,
+		)
+	);
+
+	// Calculate the percentage of learners who have completed the course, rounded off to 1 decimal point
+	$percent_complete = round( ( $completions / $learners * 100 ), 1 );
+	**/
+
+	// Get the average grade scross all learners
+	$average_grade = $course_service->get_courses_average_grade( array( $course_id ) );
+
+	// Get the average number of days it takes to complete a course
+	$average_days = $course_service->get_average_days_to_completion( array( $course_id ) );
+
+	// Set up array of data to be used
+	$data = array(
+		'learners' => array(
+			'label' => __( 'Number of learners', 'wporg-learn' ),
+			'value' => $learners,
+		),
+		/**
+		'completions' => array(
+			'label' => __( 'Completion rate', 'wporg-learn' ),
+			'value' => $percent_complete . '%',
+		),
+		**/
```

Comment:
Again I would remove this if we don't want it, but it can be fixed with
```suggestion
		
		/*
		'completions' => array(
			'label' => __( 'Completion rate', 'wporg-learn' ),
			'value' => $percent_complete . '%',
		),
		*/
```